packages/@lwc/ssr-runtime/src/lightning-element.ts

```diff
@@ -91,35 +76,31 @@ export class LightningElement implements PropsAvailableAtConstruction {
         return (this.#classList = new ClassList(this));
     }
 
-    #setAttribute(attrName: string, attrValue: string | null): void {
-        this.#attrs[attrName] = attrValue;
-        mutationTracker.add(this, attrName);
-    }
-
-    setAttribute(attrName: string, attrValue: unknown): void {
-        this.#setAttribute(attrName, String(attrValue));
+    setAttribute(attrName: unknown, attrValue: unknown): void {
+        if (typeof attrName === 'string') {
+            const normalizedValue = String(attrValue);
+            this.#attrs[attrName] = normalizedValue;
+            reflectAttrToProp(this, attrName, normalizedValue);
+            mutationTracker.add(this, attrName);
+        }
     }
 
     getAttribute(attrName: unknown): string | null {
-        if (this.hasAttribute(attrName)) {
-            return this.#attrs[attrName as string];
+        if (typeof attrName === 'string' && hasOwnProperty.call(this.#attrs, attrName)) {
+            return this.#attrs[attrName];
         }
         return null;
```

Comment:
```suggestion
        return attrName in this.#attrs ? this.#attrs[attrName] : null;
```

Reply:
Trying to avoid returning inherited properties

Reply:
I was gonna say don't we `Object.create(null)` for `attrs`, but then I remembered we don't control the top-level `attrs` object. Lame!